src/k8s/pkg/k8sd/app/hooks_post_refresh.go

```diff
@@ -24,5 +25,79 @@ func (a *App) postRefreshHook(ctx context.Context, s state.State) error {
 		return fmt.Errorf("failed to set snapd configuration from k8sd: %w", err)
 	}
 
+	k8sClient, err := a.snap.KubernetesClient("")
+	if err != nil {
+		return fmt.Errorf("failed to get Kubernetes client: %w", err)
+	}
+
+	upgrade, err := k8sClient.GetInProgressUpgrade(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get in progress upgrade: %w", err)
+	}
+
+	if upgrade != nil {
```

Comment:
You can remove a level of indentation with a guard clause here
```suggestion

	if upgrade == nil {
	   log.Info()
	   return
    }
    ...
```

Reply:
I also thought about this but if feels a bit dangerous. If people later add upgrade-unrelated steps below this logic then this logic is silently not executed if no update is in progress.